src/vs/platform/terminal/node/terminalProcess.ts

```diff
@@ -341,10 +345,16 @@ export class TerminalProcess extends Disposable implements ITerminalChildProcess
 		}, WRITE_INTERVAL_MS);
 	}
 
-	private _doWrite(): void {
+	private _doWrite(isBinary?: boolean): void {
+		console.info('writing binary', isBinary);
 		const data = this._writeQueue.shift()!;
-		this._logService.trace('IPty#write', `${data.length} characters`);
-		this._ptyProcess!.write(data);
+		if (isBinary) {
+			this._logService.info('IPty#writeBinary', `${data.length} characters`);
+			this._ptyProcess!.write(Buffer.from(data, 'binary').toString('binary'));
```

Comment:
@jerch does this look right for writing binary to node-pty (on all platforms)?

Reply:
I fear not, given that the event data is a proper bytestring (only containing codepoints in 0..255), it basically sends the input string unchanged. NodeJS will always try to do a string-UTF8 conversion at OS barriers for string types.

The only proper fix I can think of would be to send the binary buffer directly skipping NodeJS' attempt to convert anything:
```typescript
this._ptyProcess!.write(Buffer.from(data, 'binary'));
```

If thats not supported (I think `node-pty`'s conpty integration only supports string currently?), there is still a possibility to get at least lower coords through:

```typescript
function isAscii(data: string): boolean {
  return Array.from(data).every(el => el.charCodeAt(0) < 0x80)
}
...
// only support coords in ASCII range for now:
if (isAscii(data)) {
  this._ptyProcess!.write(data);
}
```